scripts/api_diff_report/api_diff_report.py

```diff
@@ -0,0 +1,259 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import argparse
+import logging
+import os
+import api_info
+import datetime
+import pytz
+
+STATUS_ADD = 'ADDED'
+STATUS_REMOVED = 'REMOVED'
+STATUS_MODIFIED = 'MODIFIED'
+STATUS_ERROR = 'BUILD ERROR'
+
+
+def main():
+    logging.getLogger().setLevel(logging.INFO)
+
+    args = parse_cmdline_args()
+
+    merged_branch = os.path.expanduser(args.merged_branch)
+    base_branch = os.path.expanduser(args.base_branch)
+    new_api_json = json.load(
+        open(os.path.join(merged_branch, api_info.API_INFO_FILE_NAME)))
+    old_api_json = json.load(
+        open(os.path.join(base_branch, api_info.API_INFO_FILE_NAME)))
+
+    diff = generate_diff_json(new_api_json, old_api_json)
+    if diff:
+        logging.info(f'json diff: \n{json.dumps(diff, indent=2)}')
+        logging.info(f'plain text diff report: \n{generate_text_report(diff)}')
+        logging.info(
+            f'markdown diff report title: \n{generate_markdown_title(args.commit, args.run_id)}'
+        )
+        logging.info(
+            f'markdown diff report: \n{generate_markdown_report(diff)}')
+    else:
+        logging.info('No API Diff Detected.')
+
+
+def generate_diff_json(new_api, old_api, level='module'):
+    """diff_json only contains module & api that has a change.
+
+    format:
+    {
+      $(moduel_name_1): {
+        "api_types": {
+          $(api_type_1): {
+            "apis": {
+              $(api_1): {
+                "declaration": [
+                  $(api_1_declaration)
+                ],
+                "sub_apis": {
+                  $(sub_api_1): {
+                    "declaration": [
+                      $(sub_api_1_declaration)
+                    ]
+                  },
+                },
+                "status": $(diff_status)
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+    NEXT_LEVEL = {
+        'module': 'api_types',
+        'api_types': 'apis',
+        'apis': 'sub_apis'
+    }
+    next_level = NEXT_LEVEL.get(level)
+
+    diff = {}
+    for key in set(new_api.keys()).union(old_api.keys()):
+        # Added API
+        if key not in old_api:
+            diff[key] = new_api[key]
+            diff[key]['status'] = STATUS_ADD
+            if diff[key].get('declaration'):
+                diff[key]['declaration'] = [STATUS_ADD
+                                            ] + diff[key]['declaration']
+        # Removed API
+        elif key not in new_api:
+            diff[key] = old_api[key]
+            diff[key]['status'] = STATUS_REMOVED
+            if diff[key].get('declaration'):
+                diff[key]['declaration'] = [STATUS_ADD
+                                            ] + diff[key]['declaration']
+        # Moudle Build Error. If a "module" exist but have no content (e.g. doc_path), it must have a build error.
```

Comment:
nit: typo and long line
```suggestion
        # Module Build Error. If a "module" exist but have no
        # content (e.g. doc_path), it must have a build error.
```

Reply:
https://github.com/firebase/firebase-ios-sdk/pull/11184/files#diff-77b8bc72e130b845bf0ad12b30ef21d68de8d64d17537e3e0879d8feabfba5c3R99
e.g. "${module}/index.html"